src/world/action/land.c

```diff
@@ -1,9 +1,206 @@
+        }
+
+        currentCollisionStatus->lastTouchedFloor = -1;
+    }
+
+    playerStatus->fallState = (u8) playerStatus->fallState + 1;
+    playerStatus->currentSpeed *= 0.6f;
+
+    player_input_to_move_vector(&inputMoveAngle, &inputMoveMagnitude);
+    check_input_jump();
+
+    if (inputMoveMagnitude != 0.0f) {
+        playerStatus->targetYaw = inputMoveAngle;
+    }
+    
+    update_locomotion_state();
+}
+
+void func_802B62CC_E24BEC(void) {
+    CollisionStatus* currentCollisionStatus = &gCollisionStatus;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    f32 inputMoveMagnitude;
+    f32 inputMoveAngle;
+    s32 playerFlagsModified;
+    s32 squaredStick0;
+    s32 squaredStick1;
+
+    if (playerStatus->flags < 0) {
+        playerFlagsModified = playerStatus->flags & 0x7FFFFFFF;
+        playerStatus->flags = playerFlagsModified;
+        playerStatus->fallState = 0;
+        playerStatus->decorationList = 0;
+        playerStatus->unk_C2 = 0;
+        playerStatus->flags = playerFlagsModified & ~0xE;
+        playerStatus->decorationPos[0] = playerStatus->position.x;
+        playerStatus->decorationPos[1] = playerStatus->position.z;
+
+        sfx_play_sound_at_player(0x148, 0);
+
+        if (((u16) currentCollisionStatus->currentFloor & 0x4000) == 0) {
+            phys_adjust_cam_on_landing();
+        }
+
+        currentCollisionStatus->lastTouchedFloor = -1;
+    }
+
+    playerStatus->fallState = (u8) playerStatus->fallState + 1;
+    playerStatus->currentSpeed *= 0.6f;
+
+    player_input_to_move_vector(&inputMoveAngle, &inputMoveMagnitude);
+
+    if (inputMoveMagnitude != 0.0f) {
+        
+        if (inputMoveMagnitude != 0.0f) {
+            playerStatus->targetYaw = inputMoveAngle;
+        }
+
+        squaredStick0 = SQ(playerStatus->stickAxis[0]);
+        squaredStick1 = SQ(playerStatus->stickAxis[1]);
+
+        if (!(sqrtf(squaredStick0 + squaredStick1) > 55.0f)) {
+            set_action_state(ACTION_STATE_WALK);
+            return;
+        }
+
+        set_action_state(ACTION_STATE_RUN);
+    } else {
+        set_action_state(ACTION_STATE_IDLE);
+    }
+}
+
+void func_802B644C_E24D6C(void) {
+    CollisionStatus* currentCollisionStatus = &gCollisionStatus;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    f32 inputMoveMagnitude;
+    f32 inputMoveAngle;
+    s32 squaredStick0;
+    s32 squaredStick1;
+    s32 playerFlagsModified;
+    
+    if (playerStatus->flags < 0) {
+        playerFlagsModified = playerStatus->flags & 0x7FFFFFFF;
+        playerStatus->flags = playerFlagsModified;
+        playerStatus->fallState = 0;
+        playerStatus->decorationList = 0;
+        playerStatus->unk_C2 = 0;
+        playerStatus->flags = playerFlagsModified & ~0xE;
+        playerStatus->decorationPos[0] = playerStatus->position.x;
+        playerStatus->decorationPos[1] = playerStatus->position.z;
+
+        if (( currentCollisionStatus->currentFloor & 0x4000) == 0) {
```

Comment:
Can you remove the space before currentCollisionstatus... here? Also, flag checks should be done as discussed in the last PR :)